Doc/library/test.rst

```diff
@@ -673,7 +1234,162 @@ The :mod:`test.support` module defines the following classes:
+
+   A context manager to temporarily add directories to sys.path.
+
+   This makes a copy of :data:`sys.path`, appends any directories given
+   as positional arguments, then reverts :data:`sys.path` to the copied
+   settings when the context ends.
+
+   Note that *all* :data:`sys.path` modifications in the body of the
+   context manager, including replacement of the object,
+   will be reverted at the end of the block.
+
+
+.. class:: SaveSignals()
+
+   Class to save and restore signal handlers registered by the Python signal
+   handler.
+
+
+.. class:: Matcher()
+
+   .. method:: matches(self, d, **kwargs)
+
+      Try to match a single dict with the supplied arguments.
+
+
+   .. method:: match_value(self, k, dv, v)
+
+      Try to match a single stored value (*dv*) with a supplied value (*v*).
+
+
 .. class:: WarningsRecorder()
 
    Class used to record warnings for unit tests. See documentation of
    :func:`check_warnings` above for more details.
+
+
+.. class:: BasicTestRunner()
+
+   .. method:: run(test)
+
+      Run *test* and return the result.
+
+
+.. class:: TestHandler(logging.handlers.BufferingHandler)
+
+   Class for logging support.
+
+
+:mod:`test.support.script_helper` --- Utilities for the Python execution tests
+==============================================================================
+
+.. module:: test.support.script_helper
+   :synopsis: Support for Python's script execution tests.
+
+
+The :mod:`test.support.script_helper` module provides support for Python's
+script execution tests.
+
+.. function:: interpreter_requires_environment()
+
+   Return ``True`` if ``sys.executable interpreter`` requires environment
+   variables in order to be able to run at all.
+
+   This is designed to be used with ``@unittest.skipIf()`` to annotate tests
+   that need to use an ``assert_python*()`` function to launch an isolated
+   mode (``-I``) or no environment mode (``-E``) sub-interpreter process.
+
+   A normal build & test does not run into this situation but it can happen
+   when trying to run the standard library test suite from an interpreter that
+   doesn't have an obvious home with Python's current home finding logic.
+
+   Setting :envvar:`PYTHONHOME` is one way to get most of the testsuite to run
+   in that situation.  :envvar:`PYTHONPATH` or :envvar:`PYTHONUSERSITE` are
+   other common environment variables that might impact whether or not the
+   interpreter can start.
+
+
+.. function:: run_python_until_end(*args, **env_vars)
+
+   Set up the environment based on *env_vars* for running the interpreter
+   in a subprocess.  The values can include ``__isolated``, ``__cleavenv``,
+   and ``TERM``.
+
+
+.. function:: assert_python_ok(*args, **env_vars)
+
+   Assert that running the interpreter with *args* and optional environment
+   variables *env_vars* succeeds (``rc == 0``) and return a ``(return code,
+   stdout, stderr)`` tuple.
+
+   If the ``__cleanenv`` keyword is set, *env_vars* is used as a fresh
```

Comment:
Since this is a keyword argument name, it should be formatter as `*__cleanenv*`.